internal/validation/apiextensions/v1/composition/handler.go

```diff
@@ -50,22 +44,21 @@ const (
 	crdsIndexKey = "crd.kind.group"
 )
 
-// handler implements the admission handler for Composition.
-type handler struct {
-	reader  client.Reader
-	decoder *admission.Decoder
-	options controller.Options
-}
+// Error strings.
+const (
+	errNotComposition = "supplied object was not a Composition"
+	errUnexpectedOp   = "unexpected operation"
+	errValidationMode = "cannot get validation mode"
 
-// InjectDecoder injects the decoder.
-func (h *handler) InjectDecoder(decoder *admission.Decoder) error {
```

Comment:
This no longer works with controller-runtime v0.15.

Reply:
yes, this and the whole implementation of the Handler can be removed, it was only needed because we weren't able to use the CustomValidator until https://github.com/kubernetes-sigs/controller-runtime/issues/1896 was merged and released as said a few line below.